hystrix-contrib/hystrix-javanica/src/main/java/com/netflix/hystrix/contrib/javanica/aop/aspectj/HystrixCommandAspect.java

```diff
@@ -260,14 +265,26 @@ private static Method getAjcMethodFromTarget(JoinPoint joinPoint) {
     }
 
 
-    private static Class<?> getGenericParameter(Type type) {
-        Type tType = ((ParameterizedType) type).getActualTypeArguments()[0];
-        String className = tType.toString().split(" ")[1];
-        try {
-            return Class.forName(className);
-        } catch (ClassNotFoundException e) {
-            throw Throwables.propagate(e);
+    private static Class<?> getFirstGenericParameter(Type type) {
+        return getFirstGenericParameter(type, 1);
+    }
+
+    private static Class<?> getFirstGenericParameter(final Type type, final int nestedDepth) {
+        int cDepth = 0;
+        Type tType = type;
+
+        for (int cDept = 0; cDept < nestedDepth; cDept++) {
+            if (!(tType instanceof ParameterizedType))
+                throw new IllegalStateException(String.format("Sub type at nesting level %d of %s is expected to be generic", cDepth, type));
+            tType = ((ParameterizedType) tType).getActualTypeArguments()[0];
```

Comment:
I think it should be: ``` tType = ((ParameterizedType) tType).getActualTypeArguments()[cDept] ```
BTW: In a separate branch I'm working on strategic solution to compare generic types. Will refactor this code to use the common feature.